### PR TITLE
Start replacing os.path with pathlib

### LINKED ISSUE #258: Pathlib instead of os.path

### PR DESCRIPTION
**Pull request by shubhanshu02**
Fixes #258

## Description

This PR proposes the change of style to use the Pathlib styling for joining the paths instead of os.path.join().

### CODE REVIEW

File: src/gnn_tracking/preprocessing/point_cloud_builder.py [hunk 1]
```diff
@@ -57,9 +56,9 @@ def __init__(
         """
         # create outdir if necessary
         os.makedirs(outdir, exist_ok=True)
-        self.outdir = outdir
```
Comment (klieret) [inline src/gnn_tracking/preprocessing/point_cloud_builder.py:60]: I'd say this should be `Path(..)` (there's no reason not to have writing support here)

**Review by klieret — CHANGES_REQUESTED** [review]

**Comment by klieret** [comment]
Looks very solid, except for the one comment I had. Thanks a lot! Let's wait for the tests :)

**Comment by klieret** [comment]
Hi: As you can see some of the tests still fail because of this bit here:

```
evtid = int(f[-9:])
```

(assuming a string). Arguably this is a hacky thing to begin with, but we could probably just replace it with something like `f.name[-9:])` for now

File: src/gnn_tracking/preprocessing/point_cloud_builder.py (hunk 1 above)
Comment (shubhanshu02) [reply src/gnn_tracking/preprocessing/point_cloud_builder.py:60]: I noticed in the function declaration that `outdir: str | PurePath`. `PurePath` seems to be a more specific type of `PathLike` objects, so I used the following:

```py
self.outdir = PurePath(outdir)
```

**Comment by shubhanshu02** [comment]
There's one thing I'm concerned here with changing the following to `f.name`.
```py
evtid = int(f[-9:])
```

Which one of the following should be used here?
- `PurePath.name` gives output just the final path component i.e. name of the file, not the entire path. 
- `str(path)` converts the entire path to a path string.

File: src/gnn_tracking/preprocessing/point_cloud_builder.py (hunk 1 above)
Comment (klieret) [reply src/gnn_tracking/preprocessing/point_cloud_builder.py:60]: no, `Path` is a subclass of `PurePath` (see [docs](https://docs.python.org/3/library/pathlib.html)), so it is more specific than the latter. We want to accept a broad range of inputs (actually `outdir: str| os.PathLike` would've been even better ) and then convert it to a `Path`

**Comment by klieret** [comment]
`Path.name` is the only thing that makes sense here

**Review by klieret — CHANGES_REQUESTED** [review]

File: src/gnn_tracking/preprocessing/point_cloud_builder.py (hunk 1 above)
Comment (shubhanshu02) [reply src/gnn_tracking/preprocessing/point_cloud_builder.py:60]: Sure. I'll update.

**Review by klieret — APPROVED** [review]
Thanks again!  I will merge this. If you want to continue working on this, there are still some `os.` methods in here (e.g., `os.makedirs` that can be translated to their pathlib alternative)

**Comment by shubhanshu02** [comment]
Ya sure, I would like to work further. If I understood correctly, do you mean changing these - `os.makedirs` and `os.listdir` to their `Pathlib` alternatives?

**Pull request merged**